ffi/diplomat/src/lib.rs

```diff
@@ -44,40 +44,68 @@ extern crate std as rust_std;
 
 extern crate alloc;
 
+#[cfg(feature = "icu_properties")]
 pub mod bidi;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod calendar;
+#[cfg(feature = "icu_collator")]
 pub mod collator;
 pub mod common;
 pub mod data_struct;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod date;
+#[cfg(any(feature = "icu_datetime"))]
 pub mod datetime;
+#[cfg(feature = "icu_datetime")]
 pub mod datetime_formatter;
+#[cfg(feature = "icu_decimal")]
 pub mod decimal;
+#[cfg(feature = "icu_displaynames")]
 pub mod displaynames;
 pub mod errors;
 pub mod fallbacker;
+#[cfg(feature = "icu_decimal")]
 pub mod fixed_decimal;
+#[cfg(feature = "icu_list")]
 pub mod list;
 pub mod locale;
+#[cfg(feature = "icu_locid_transform")]
 pub mod locid_transform;
 pub mod logging;
+#[cfg(feature = "icu_normalizer")]
 pub mod normalizer;
+#[cfg(feature = "icu_normalizer")]
 pub mod normalizer_properties;
+#[cfg(feature = "icu_plurals")]
 pub mod pluralrules;
+#[cfg(feature = "icu_properties")]
 pub mod properties_maps;
+#[cfg(feature = "icu_properties")]
 pub mod properties_names;
+#[cfg(feature = "icu_properties")]
 pub mod properties_sets;
+#[cfg(feature = "icu_properties")]
 pub mod properties_unisets;
 pub mod provider;
```

Comment:
Nit: break these `mod`s into two blocks, one for components with feature gates, one for core diplomat logic